src/apps/chem/xcfunctional_libxc.cc

```diff
@@ -137,6 +140,145 @@ void XCfunctional::initialize(const std::string& input_line, bool polarized,
     }
 }
 
+void XCfunctional::set_external_parameters_if_a_functional_has_been_specified(std::stringstream &line) {
+    if (funcs.empty()) {
+        throw std::runtime_error("A functional must be specified before specifying the external parameters!");
+    }
+    else {
+        set_external_parameters(line);
+    }
+}
+
+void XCfunctional::set_external_parameters(std::stringstream &line) {
+    xc_func_type* current_functional = funcs.back().first;
+    auto parameters_name_value_pair = get_external_parameter_name_and_value_pair(current_functional);
+
+    line.ignore();
+    if (line.get() == '{') {
+        bool found_closing_brace = false;
+        while (!found_closing_brace) {
+            std::string dictionary_key;
+            std::string dictionary_value_string;
+            if (line >> dictionary_key && line >> dictionary_value_string) {
+                format_dictionary_key_if_input_ends_in_colon(dictionary_key);
+                const double dictionary_value = get_dictionary_value_if_input_ends_with_coma_or_closing_brace(
+                        dictionary_value_string, found_closing_brace);
+
+                parameters_name_value_pair = get_updated_external_parameters_name_value_pair_if_key_exists(
+                        parameters_name_value_pair, dictionary_key, dictionary_value);
+            }
+        }
+        xc_func_set_ext_params(current_functional, parameters_name_value_pair.second.data());
+    }
+    else {
+        throw std::runtime_error("The external parameters must be defined as a dictionary!");
+    }
+}
+
```

Comment:
As the following additions are all helper functions to process the input string: Maybe move at the end of the file and add a comment block before, explaining that everything that follows is just that